openedx/frontend-app-course-authoring
# feat: adds sort widget to search manager and library component page [FC-0059]

## Description

Adds a widget to the library component search manager which sorts the search results based on the selected option.

![image](https://github.com/openedx/frontend-app-course-authoring/assets/7556571/b65c7474-39c7-4e50-a02f-789e5834e641)

Plus the Recently Modified components section from https://github.com/open-craft/frontend-app-course-authoring/pull/52.

## Supporting information

Part of https://github.com/openedx/frontend-app-course-authoring/issues/1038

Blocked by: 
* https://github.com/openedx/edx-platform/pull/35103
* https://github.com/openedx/frontend-app-course-authoring/pull/1141

Private-ref: [FAL-3758](https://tasks.opencraft.com/browse/FAL-3758)

## Testing instructions

Run the latest `master` branch in Studio/CMS

0. Reindex studio to update the index settings: `tutor dev run cms ./manage.py cms reindex_studio --experimental`
1. Create a new library
2. Add components to the library.
3. From the shell, publish the library:

   ```
   from openedx.core.djangoapps.content_libraries import api, models
   library = models.ContentLibrary.objects.get(slug='<your library slug>')
   api.publish_changes(library.library_key)
   ```
4. Add more components, but leave them unpublished.

To test this change:

1. Navigate to the new library from http://apps.local.edly.io:2001/course-authoring/libraries/
2. Check that the Recently Modified section shows your recently modified components in order of most recent.
4. Change the sort options and note changes in the Components list.
   The Recently Modified section should not change when sort options change.
5. Click on the Components tab, and note sort option stays the same as previously selected.
6. Change sort options and note changes in the components list. 
7. Select "Recently Published" -- should omit unpublished components from the list.
